### DISCUSSION TITLE
Integrating with another payment gateway

### DISCUSSION THREAD

**Discussion by IanvanZyl636** [payloadcms/payload#4824] [General]
# Documentation Issue

I reside in South Africa and Stripe is not officially supported in South Africa. So I would like to integrate a new payment gateway. Currently, I could not find a way to integrate with any other payment gateways there are some mentions of hooks but it does not give you an in-depth set of instructions on how one would go about integrating it.

Any assistance would be greatly appreciated!

**Comment by elektropay** [comment]
I'm wondering the same thing also, I just subscribed to this thread.

**Reply by mommaroodles** [reply]
Me too - I did find Payfast payment gateway for Next.Js here- but i havent looked at it.

**Comment by paulpopus** [comment]
Hey guys, using the [Ecommerce Template](https://github.com/payloadcms/payload/tree/main/templates/ecommerce) as an example, I'll explain what's going on with Stripe and how you could replace it

Stripe in this case is used for a few kinds of integrations, one is that Stripe contains some of the information about your products, so we sync that with our Payload content and we do that for users too.

I recommend reading through the endpoints in `payload/endpoints` in that template as it kind of contains all the logic that the frontend is integrating with.

The other section containing Stripe API logic is inside the collections' hooks such as `collections/Products/hooks` where products are sync'd and updated.

And lastly you have webhooks to handle updates on Payload's side once checkouts (or subscriptions for example) and other events are completed on the processor's side.

Other payment processors work in a similar way such as LemonSqueezy, where you sync some of your data or at least connect them and then you can create invoices or paymentIntents/checkouts or whatever name they have through their API.

It's obviously quite a bit more complex than I can cover in one comment and I think we discussed either an adapter method or updating examples for ecommerce in this [discussion thread](https://github.com/payloadcms/payload/discussions/4352).

**Comment by paulpopus** [comment]
I think this is better suited for a discussion thread instead

**Comment by anupamavm** [comment]
I also in need to integrate another payment gateway other than stripe. Is anyone has clear guide that i can follow

**Comment by jpmelnik** [comment]
Me too. I want to integrate a custom payment gateway. I've just started reading and would like to know how to integrate Mercado Pago and PayPal.

**Comment by maximpokrovski** [comment]
I’ve been in a similar situation before, and after testing several options, I ended up preferring [Paykassma](https://paykassma.com/business-solutions/payment-gateway-for-freelancers) it’s much easier to integrate, supports freelancers and small businesses globally, and works smoothly even in regions where Stripe isn’t available.

**Comment by paulpopus** [comment]
Closing this discussion as we now support a [payment adapter pattern](https://payloadcms.com/docs/ecommerce/payments) in the ecommerce plugin, if you have any feedback regarding that you can post it in the [new discussion](https://github.com/payloadcms/payload/discussions/14239)
